design-editor/src/panel/wizards/page-wizard-element.js

```diff
@@ -23,131 +23,118 @@ const TEMPLATE_PATH = '/panel/wizards/page-wizard-element.html',
+class PageWizard extends Component {
+	/**
+     * Responsible for new Page wizard component
      */
-    _parseCategories(templateMetadata) {
-        var data, metadata = templateMetadata._packages;
+	constructor() {
+		super();
+		if (!this.initialized) {
+			this._initialize();
+			this.events = {
+				'click .pw-accept-btn': 'onAccept',
+				'click .pw-cancel-btn': 'onCancel'
+			};
+
+			this._template = '';
+		}
+	}
 
-        Object.keys(metadata).forEach((templateName) => {
-            data = $.extend(true, {}, metadata[templateName].options);
-            data.id = templateName;
-            data.templateFileName = data.templateFileName || DEFAULT_TEMPLATE_NAME;
+	/**
+	 * Initialize components state
+	 */
+	_initialize() {
+		this._appPath = appManager.getAppPath();
+		this._templateMetadata = {};
 
-            if (!this._templateMetadata[data.profile]) {
-                this._templateMetadata[data.profile] = [];
-            }
+		this._pageNameInputElement = null;
+		this._pagePathInputElement = null;
+		this._categoryElement = null;
+		this._thumbnailElement = null;
 
-            this._templateMetadata[data.profile].push(data);
-        });
-    }
+		this._newPageInfo = {};
+		this._parseCategories(packageManager.getPackages(Package.TYPE.PAGE_TEMPLATE));
+		this.setAttribute('tabindex', '-1');
 
-    /**
-     * Open page wizard
-     */
-    openPageWizard() {
-        var self = this,
-            projectPath = pathUtils.createProjectPath();
+		eventEmitter.on(EVENTS.PageWizardOpen, this.openPageWizard.bind(this));
+	}
 
-        this._projectPath = projectPath;
-        this._projectProfile = StateManager.get('screen').profile;
+	/**
+	 * Change template view
+	 * @param {string} profileName
+	 */
+	_changeTemplateView(profileName) {
+		this._thumbnailRefresh(this._templateMetadata[profileName]);
+	}
 
-        $.ajax({
-            url: path.join(self._appPath.src, TEMPLATE_PATH)
-        }).done((html) => {
-            self.$el.html(html);
-            eventEmitter.emit(EVENTS.OpenPanel, {
-                type: 'modal',
-                item: self
-            });
+	/**
+	 * Parse categories
+	 * @param {Object} templateMetadata
+	 */
+	_parseCategories(templateMetadata) {
+		const metadata = templateMetadata._packages;
 
-            self._pagePathInputElement = self.$el.find('closet-file-picker')[0];
-            self._pageNameInputElement = self.$el.find('.pw-page-name')[0];
-            self._pageTemplateWrapperElement = self.$el.find('.closet-template-thumbnail-wrapper')[0];
-            self._thumbnailElement = self.$el.find('closet-pw-thumbnail')[0];
+		Object.keys(metadata).forEach((templateName) => {
+			const data = $.extend(true, {}, metadata[templateName].options);
+			data.id = templateName;
+			data.templateFileName = data.templateFileName || DEFAULT_TEMPLATE_NAME;
 
-            self._pagePathInputElement.path = projectPath;
+			if (!this._templateMetadata[data.profile]) {
```

Comment:
It can be rewritten with spread operator (to make it more readable)
```
const data = { 
   ...metadata[templateName].options, 
   id: templateName, 
   templateFileName: data.templateFileName || DEFAULT_TEMPLATE_NAME
}
```